browser/ui/ai_chat/ai_chat_tab_helper_unittest.cc

```diff
@@ -457,4 +479,107 @@ TEST_P(AIChatTabHelperUnitTest,
   }
 }
 
+#if BUILDFLAG(ENABLE_PDF)
+TEST_P(AIChatTabHelperUnitTest, OnAllPDFPagesTextReceived) {
+  // Create test data with out-of-order pages
+  std::vector<std::pair<size_t, std::string>> page_texts = {
+      {2, "Page 3 content"},
+      {0, "Page 1 content"},
+      {1, "Page 2 content"},
+  };
+
+  base::test::TestFuture<std::string, bool, std::string> future;
+  OnAllPDFPagesTextReceived(future.GetCallback(), page_texts);
+
+  auto [content, is_video, invalidation_token] = future.Get();
+  EXPECT_FALSE(is_video);
+  EXPECT_TRUE(invalidation_token.empty());
+  EXPECT_EQ(content, "Page 1 content\nPage 2 content\nPage 3 content");
+}
+
+TEST_P(AIChatTabHelperUnitTest, OnGetPDFPageCount_FailedStatus) {
+  base::test::TestFuture<std::string, bool, std::string> future;
+
+  OnGetPDFPageCount(future.GetCallback(),
+                    pdf::mojom::PdfListener::GetPdfBytesStatus::kFailed,
+                    std::vector<uint8_t>(), 0);
+
+  auto [content, is_video, invalidation_token] = future.Get();
+  EXPECT_TRUE(content.empty());
+  EXPECT_FALSE(is_video);
+  EXPECT_TRUE(invalidation_token.empty());
+}
+
+TEST_P(AIChatTabHelperUnitTest, OnGetPDFPageCount_Success) {
```

Comment:
Maybe worth adding NoHelper to the test name to avoid confusion.

Reply:
added in df16b1431f43f95bae92aa83ba19e8de968beb03